### PR TITLE
Fixed contact menu position on comments

### LINKED ISSUE #7250: Wrong position of the contacts menu for comment authors
In the _Comments_ app the contacts menu is shown when clicking on the name of a comment author (and which is not the current user). However, since commit 2348d14e48d2482343bcc111347efcdd53c0faf8 from #7051 (great work, by the way ;-) ), the menu appears in a wrong position.

@skjnldsv

Before commit:
contactsmenu-before

After commit:
contactsmenu-after

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #7250 

The popover needs a fix too for the activity app, but this needs to be done on the activity repo.

### CODE REVIEW

**Comment by danxuliu** [comment]
Unfortunately setting the `top` CSS attribute fixed the contacts menu when shown for a comment author, but moved it when shown for a mention in the text:

Author:
contactsmenu-author

Mention:
contactsmenu-message

I have investigated the issue and it turns out that the commit 2348d14e48d2482343bcc111347efcdd53c0faf8 did not break the position; the vertical position was already wrong, that commit just made it more evident ;-)

The real cause was displaying the child `div`s of `.authorRow` as `inline-block`; this caused the `.contactsmenu-popover` to be displayed too as `inline-block`, which in turn caused the top of the contacts menu to be aligned to the top of the author row. Before the commit 2348d14e48d2482343bcc111347efcdd53c0faf8 `.menu` elements had a `top: 45px` rule, which moved the contacts menu to the lower part of the author row (in fact it was too low), and when that commit removed the general CSS rules for `.menu` by assigning them to `#header .menu` then there was no vertical displacement for the contacts menu and thus it appeared aligned to the top of the author row.

I must admit, however, that I do not understand why displaying the contacts menu popover as a block aligns it to the center of the avatar (I can not see any `top: 32px` rule applied to the contacts menu), but it works :-P , so if someone could clarify that it would be great ;-)

Anyway I took the liberty to amend your commit to just fix the horizontal position and add another commit to fix the vertical position by displaying the contacts menu as a block; I did that using the `:not` selector, but I do not know if the preferred approach is to use that selector or just add a different, more specific rule that reverts the general one (`#commentsTabView .authorRow>div.contactsmenu-popover { display: block; }`).

**Comment by skjnldsv** [comment]
@danxuliu there, I fixed most of the stuff.
We definitely needs to standardize the sidebars, this is a mess! :'(
cc @jancborchardt

**Comment by pixelipo** [comment]
Yes, @skjnldsv - check issue https://github.com/nextcloud/server/issues/7255

**Review by pixelipo — COMMENTED** [review]
image

I'm not really sure what causes these errors, but they should be fixed.

File: apps/comments/css/comments.css [hunk 1]
```diff
@@ -126,6 +127,13 @@
 #commentsTabView .comment .authorRow {
 	position: relative;
 	cursor: pointer;
+	display: inline-flex;
+	align-items: center;
+	width: 100%;
+}
+
+#commentsTabView .comments li .message .atwho-inserted {
+	margin-left: 5px;
```
Comment (pixelipo) [inline apps/comments/css/comments.css]: I think this should be applied to `.avatar-name-wrapper` instead.

**Comment by danxuliu** [comment]
@skjnldsv 
>  Sorry I did not realised there was a popover for the main avatar as well :)

No problem ;-)

>  there, I fixed most of the stuff.

Thanks :-) But... can I ask what was wrong with the fix that I pushed to this branch?

Also, from what I see with your new changes the tip of the arrow of the tooltip is no longer at the centre of the avatar, but a little to the right and to the bottom (although maybe that is intended, I do not know).

**Comment by skjnldsv** [comment]
> But... can I ask what was wrong with the fix that I pushed to this branch?

I'm an idiot, I overwrite it with my push! 🙈

**Comment by danxuliu** [comment]
>  I'm an idiot, I overwrite it with my push!

xD To be honest I thought it was something like that :-P Do not worry; I can rebase your new changes on mines or push to a different branch so you can compare. As you prefer ;-)

**Comment by skjnldsv** [comment]
@danxuliu don't mock me! :p 
Do as you please, forgive me :)

**Comment by danxuliu** [comment]
@skjnldsv
>  Do as you please, forgive me :)

You have my blessing :-P

I have rebased your last commit onto my previous changes; due to my changes it seems that your last commit is not needed to fix the contact menu position, but I prefer you to check it. Also, even if it was not needed to fix the position it may cover other cases I am not aware of, or simply enhance the CSS rules (I am aware of but not very familiar with flex displays, so I do not know its implications here). In any case the commit message should be reworded ;-)

Besides all that, from what I can see the tip of the arrow is slightly moved to the right and to the bottom. It is caused by `left: -6px` instead of `left: -7px` and `margin-top: 0` in `.contactsmenu-popover` in last commit, but as I said maybe it is an explicit design decision, I do not know.

@pixelipo 
Could you provide a screenshot with some more context (I mean, not so focused in the problem)? I do not know what I am seeing :-)

**Review by nickvergessen — APPROVED** [review]
Works

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**